astroquery/simbad/core.py

```diff
@@ -936,6 +1006,243 @@ def query_objectids_async(self, object_name, *, cache=True,
+        <Table length=5>
+         table_name column_name   datatype  ...  unit           ucd
+           object      object      object   ... object         object
+        ----------- ----------- ----------- ... ------ ----------------------
+             filter description UNICODECHAR ...        meta.note;instr.filter
+             filter  filtername     VARCHAR ...                  instr.filter
+             filter        unit     VARCHAR ...                     meta.unit
+               flux      filter     VARCHAR ...                  instr.filter
+        mesDiameter      filter        CHAR ...                  instr.filter
+        """
+        condition = f"LIKE LOWERCASE('%{_adql_parameter(keyword)}%')"
+        query = ("SELECT table_name, column_name, datatype, description, unit, ucd"
+                 " FROM TAP_SCHEMA.columns"
+                 f" WHERE (LOWERCASE(column_name) {condition})"
+                 f" OR (LOWERCASE(description) {condition})"
+                 f" OR (LOWERCASE(table_name) {condition})"
+                 " ORDER BY table_name, principal DESC, column_name")
+        if get_adql:
+            return query
+        return self.query_tap(query)
+
+    def find_linked_tables(self, table: str, get_adql=False):
+        """
+        Expose the tables that can be non-obviously linked with the given table.
+
+        This is not exhaustive, this list contains only the links where the column names
+        are not the same in the two tables. For example every ``oidref`` column of any
+        table can be joined with any other ``oidref``. The same goes for every ``otype``
+        column even if this is not returned by this method.
+
+        Parameters
+        ----------
+        table : str
+            One of SIMBAD's tables name
+        get_adql : bool, optional
+            Returns the ADQL string instead of querying SIMBAD.
+
+        Returns
+        -------
+        `~astropy.table.table.Table`
+            The information necessary to join the given table to an other.
+
+        Examples
+        --------
+        >>> from astroquery.simbad import Simbad
+        >>> Simbad.find_linked_tables("otypes") # doctest: +REMOTE_DATA
+        <Table length=2>
+        from_table from_column target_table target_column
+          object      object      object        object
+        ---------- ----------- ------------ -------------
+          otypedef       otype       otypes         otype
+            otypes      oidref        basic           oid
+        """
+        query = ("SELECT from_table, from_column, target_table, target_column"
+                 " FROM TAP_SCHEMA.key_columns JOIN TAP_SCHEMA.keys USING (key_id)"
+                 f" WHERE (from_table = '{_adql_parameter(table)}')"
+                 f" OR (target_table = '{_adql_parameter(table)}')")
+        if get_adql:
+            return query
+        return self.query_tap(query)
+
+    def query_tap(self, query: str, maxrec=10000, uploads=None):
```

Comment:
make optional kwargs kwarg only. Here, and in all the other methods. Also, we don't really do any type hinting, but it would not hurt to leave it in there.

```suggestion
    def query_tap(self, query: str, *, maxrec=10000, uploads=None):
```

Reply:
Done :)